### PR TITLE
Use django default filters instead of babel

### LINKED ISSUE #6485: Cleanup our |datetime, |babel_datetime and |babel_date template filters
* `|babel_date` is unused as far as I can tell, might need more verification
* `|babel_datetime` and `|datetime` template filters should be ported to django's `|date` template filter

Something like

```python
from django.template import defaultfilters

@register
def format_date(value, format='DATE_FORMAT'):
    return defaultfilters.date(value, format)

@register
def format_datetime(value, format='DATETIME_FORMAT'):
    return defaultfilters.date(value, format)
```

luckily we don't have any specific time filters in use as far as I can tell but that's worth investigating more too.

Or should we just use https://github.com/python-babel/django-babel#using-the-template-tags instead of Django's house tools? We are using django-babel mostly for extraction-reasons but not much beyond that. Babel's date/time formatting capabilities are highly superior to what Django provides though so maybe it's worth just using that? I usually can't decide where to start, simplicity first, use Django unless we need more powerful tools?

cc @eviljeff @diox for discussion

### PR DESCRIPTION
**Pull request by harry-7**
Please delete anything that isn't relevant to your patch.
Fixes #6485 
Remove babel_date and babel_datetime filters and use django template filters in place of it.

### CODE REVIEW

File: src/olympia/amo/templatetags/jinja_helpers.py [hunk 1]
```diff
@@ -624,17 +614,11 @@ def nl2br(string):
 
 
 @library.filter(name='datetime')
-def datetime_filter(t, fmt=None):
-    """Call ``datetime.strftime`` with the given format string.
-
-    This is a copy from jingo until we have time to port all our templates
-    to use django's datetime filter.
+def datetime_format(value, format='F j, Y'):
```
Comment (eviljeff) [inline src/olympia/amo/templatetags/jinja_helpers.py]: May as well call it `def format_datetime`, use `format='DATETIME_FORMAT'`.  And I think we're going to need a `def format_date` after-all too.

**Comment by eviljeff** [comment]
So, as mentioned in the issue, change `|datetime` usages in the templates to `|date`, and add
```
@library.filter(name='date')
def format_date(value, format='DATE_FORMAT'):
    return defaultfilters.date(value, format)
```
Then then usages of custom date strings will need to be fixed.

`|datetime(_('%%b %%e, %%Y')` to `|date`
`|datetime('%Y-%m-%d')` to `|date('Y-m-d')`
`|datetime('%B %e, %Y %H:%M:%S')` to `|datetime`
`|datetime('%Y-%m-%d %H:%M:%S')` to `|datetime('Y-m-d H-M-S')`

**Comment by harry-7** [comment]
@eviljeff . I have made requested changes. but are you sure we want to keep the default format as `DATETIME_FORMAT`? Because the behaviour of this is different from the previous behaviour of our datetime. So wanted to confirm.

**Comment by eviljeff** [comment]
(Sorry for the delay in getting back to this... and how it's turned into a more complex issue than I first thought)
> Are you sure we want to keep the default format as DATETIME_FORMAT? Because the behaviour of this is different from the previous behaviour of our datetime. So wanted to confirm.

Our `|datetime` was really defaulting to something similar to `DATE_FORMAT`, rather than `DATETIME_FORMAT` so that's why asked for `|datetime` to be changed to `|date` in most cases.  We want the output to be the default really - makes for easier maintenance if the upstream code changes (or the tests change at least).

In the meantime we actually landed a big change to /editors/ files (moved to /reviewers) so you'll need to do a rebase now.

**Comment by harry-7** [comment]
@eviljeff  I rabased those changes. made some changes to the tests so that they will match the current behaviour.

**Comment by eviljeff** [comment]
@harry-7 thanks for doing this, and sorry (again) for not following up.  thanks @EnTeQuAk for picking this up too 👍

**Pull request merged**